client/my-sites/themes/validate-filters.js

```diff
@@ -81,3 +83,33 @@ export function sortFilterTerms( context, terms ) {
 		.sort()
 		.map( ( filter ) => getThemeFilterTermFromString( context.store.getState(), filter ) );
 }
+
+export function fetchAndValidateVerticalsAndFiltersIfLoggedIn( context, next ) {
+	const state = context.store.getState();
+	const currentUser = getCurrentUser( state );
+
+	if ( ! currentUser ) {
+		return next();
+	}
+
+	return fetchThemeFilters( context, () => {
+		return validateVertical( context, () => {
+			return validateFilters( context, next );
```

Comment:
A little nitpick: the handlers don't return anything -- they pass control solely using the callbacks. You can just call `validateVertical` etc. without the `return` statement.